src/wallet/wallet.cpp

```diff
@@ -1157,12 +1157,13 @@ bool CWallet::AddToWallet(CTransactionRef tx, const UpdateWalletTxFn& update_wtx
     return true;
 }
 
-void CWallet::LoadToWallet(const CWalletTx& wtxIn)
+bool CWallet::LoadToWallet(const uint256& hash, const UpdateWalletTxFn& update_wtx)
 {
-    uint256 hash = wtxIn.GetHash();
-    const auto& ins = mapWallet.emplace(hash, wtxIn);
+    auto ins = mapWallet.emplace(std::piecewise_construct, std::forward_as_tuple(hash), std::forward_as_tuple(this, nullptr));
```

Comment:
Hmmm new execution flow doesn't make it easier to understand. Maybe mapWallet should be access directly in `ReadKeyValue`, when as you said in another comment wallet components are broken up in smaller classes. I think, but need to test, that `MarkConflicted` and its loop is useless here.